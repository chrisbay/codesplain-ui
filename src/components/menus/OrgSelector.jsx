import React, { PropTypes } from 'react'
import { DropDownMenu, MenuItem } from 'material-ui'

class OrgSelector extends React.Component {
  constructor(props) {
    super(props);
    this.state = {value: props.orgs[0]};
    this.handleChange = this.handleChange.bind(this);
  }
  handleChange(ev, tgt, value) {
    const { onChange } = this.props
    this.setState({value});
    onChange(value)
  }
  render() {
    const { value } = this.state;
    const { orgs } = this.props;
    return (
      <div>
        <DropDownMenu value={value} onChange={this.handleChange}>
          {orgs.map((org) => (
            <MenuItem
              key={org}
              value={org}
              primaryText={org}
            />
          ))}
        </DropDownMenu>
      </div>
    );
  }
}

OrgSelector.propTypes = {
  orgs: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
}

export default OrgSelector;
