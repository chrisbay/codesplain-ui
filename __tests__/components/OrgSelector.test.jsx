import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import OrgSelector from '../../src/components/menus/OrgSelector';

describe('<OrgSelector />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <OrgSelector />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
