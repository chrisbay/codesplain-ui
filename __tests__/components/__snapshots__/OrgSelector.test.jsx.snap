exports[`<OrgSelector /> matches snapshot 1`] = `
<div>
  <DropDownMenu
    animated={true}
    autoWidth={true}
    disabled={true}
    iconButton={<NavigationArrowDropDown />}
    maxHeight={500}
    multiple={false}
    onChange={[Function]}
    openImmediately={false}
    value={0}>
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="galactic-federation"
      value="galactic-federation" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="international-counsel-of-ricks"
      value="international-counsel-of-ricks" />
  </DropDownMenu>
</div>
`;
